tools/eosiocpp.in

```diff
@@ -32,8 +31,24 @@ function build_contract {
         name=`basename $file`
         filePath=`dirname $file`
 
-        echo @WASM_CLANG@ -emit-llvm -O3 --std=c++14 --target=wasm32 -ffreestanding -nostdlib -fno-threadsafe-statics -fno-rtti -fno-exceptions -I ${EOSIO_INSTALL_DIR}/include -I $filePath -c $file -o $workdir/built/$name
-        @WASM_CLANG@ -emit-llvm -O3 --std=c++14 --target=wasm32 -ffreestanding -nostdlib -fno-threadsafe-statics -fno-rtti -fno-exceptions -I ${EOSIO_INSTALL_DIR}/include -I $filePath -c $file -o $workdir/built/$name
+        # echo @WASM_CLANG@ -emit-llvm -O3 --std=c++14 --target=wasm32 -nostdinc \
+        #              -nostdlib -nostdlibinc -ffreestanding -nostdlib -fno-threadsafe-statics -fno-rtti \
+        #              -fno-exceptions -I ${EOSIO_INSTALL_DIR}/include \
+        #              -I${EOSIO_INSTALL_DIR}/include/libc++/upstream/include \
+        #              -I${EOSIO_INSTALL_DIR}/include/musl/upstream/include \
+        #              -I${BOOST_INCLUDE_DIR} \
+        #              -I $filePath \
+        #              -c $file -o $workdir/built/$name
+
+        @WASM_CLANG@ -emit-llvm -O3 --std=c++14 --target=wasm32 -nostdinc \
+                     -nostdlib -nostdlibinc -ffreestanding -nostdlib -fno-threadsafe-statics -fno-rtti \
+                     -fno-exceptions -I ${EOSIO_INSTALL_DIR}/include \
+                     -I${EOSIO_INSTALL_DIR}/include/libc++/upstream/include \
+                     -I${EOSIO_INSTALL_DIR}/include/musl/upstream/include \
+                     -I${BOOST_INCLUDE_DIR} \
```

Comment:
Including anything outside of the install directory is not good. However I don't have any better solution to suggest at the moment. We need to find a way to fix this better long term.